### ISSUE TITLE
Version compatibility issue: Unable to write any cells under golang 1.21.0.

### ISSUE THREAD

**Issue by zhengkai** [qax-os/excelize#1603]
**Description**

This morning, after an automatic upgrade to `1.21.0`, I noticed that excelize is unable to write any cells, even using the examples from the official readme. The sheets, however, appear to be functioning normally. There are no error messages during the usage of `SetCellValue` and `SaveAs`, but the cell contents remain blank.

I didn't make any code changes; I simply replaced `go build` with `go1.20.7 build` and everything returned to normal.

**Environment details (OS, Microsoft Excel™ version, physical, etc.):**
Ubuntu 22.04
LibreOffice 7.3.7.2
Intel 9700K

**Comment by xuri** [comment]
Thanks for your issue. This was similar with #1465, #1595. There are some incompatible changes in the Go 1.21 encoding/xml library.  I have given feedback to the Go team and created a patch for it, and still waiting for a reply. Please apply the patchsets in the CL https://go.dev/cl/466295 if you can.

**Comment by xijaja** [comment]
I just found this problem too。

**Comment by fifsky** [comment]
Is it compatible with go1.21? It will be a long time before the official merger. Projects that rely on excelize will not be able to use go1.21

**Comment by fifsky** [comment]
temporary solution

go.mod add line:
```
replace github.com/xuri/excelize/v2 => github.com/verystar/excelize/v2 v2.7.1
```

**Comment by xuri** [comment]
Hi @fifsky. I'll consider revendoring Go encoding/xml package in 1.16.5 (for backward compatible) in `github.com/xuri/xml` this week.

**Comment by fifsky** [comment]
Yes, I used your `github.com/xuri/xml` package replaced the `encoding/xml` see https://github.com/verystar/excelize/blob/master/go.mod#L11

**Comment by fifsky** [comment]
> Hi @fifsky. I'll consider revendoring Go encoding/xml package in 1.16.5 (for backward compatible) in `github.com/xuri/xml` this week.

OK, thank you for your reply

**Comment by xuri** [comment]
Hi @fifsky, I noticed that the Go team has added golang/go#62051 to the Go1.21.1 milestone, so I suggest waiting for the next minor releases, and you can vendoring the Go encoding/xml package by yourself if you need to.

**Comment by xuri** [comment]
> Removing the `XMLName` field from struct types that are not expected to have an explicit namespace seems to be a solution.
> 
> https://go.dev/play/p/z-acRVzkk6G
> 
> That should work as long as the type is not marshaled directly and its usage in other structs has the proper `xml` tag.

Yeah, but we also need to marshal sub-type structs variable directly (such as marshal type variable `B` to XML element `<b></b>` in the example), in this case, still need type `B` has the proper `xml.Name` tag.

**Comment by fifsky** [comment]
@xuri   I noticed that, too, thinks

**Comment by exfly** [comment]
I just found this problem too。

**Comment by davidlam-winc** [comment]
Got the same problem.  This must the first time since my Go programming that I'm forced to back a version!   Hope that we don't end up needing to keep different versions of Go!

**Comment by fnickels** [comment]
> Hi @fifsky, I noticed that the Go team has added [golang/go#62051](https://github.com/golang/go/issues/62051) to the Go1.21.1 milestone, so I suggest waiting for the next minor releases, and you can vendoring the Go encoding/xml package by yourself if you need to.

What is involved in "vendoring the Go encoding/xml package"?  Any pointers on how to attempt this?  is this a simple change to the go.mod file, or some other process?

**Comment by xuri** [comment]
I have vendoring Go 1.16.15's encoding/xml package in github.com/xuri/xml, I will use it in the Excelize instead of encoding/xml standard library tomorrow, after that, you can upgrade the Excelize library with the master branch code, and these changes will be released in the next Excelize version, if Go 1.21.1 not yet released.

**Comment by xuri** [comment]
Since vendoring Go 1.16.15 encoding/xml will introduce security issues [CVE-2022-28131](https://nvd.nist.gov/vuln/detail/CVE-2022-28131) (fixed in the go1.18.4) and [CVE-2022-30633](https://nvd.nist.gov/vuln/detail/CVE-2022-30633) (fixed in the go1.18.10), so I suggest waiting for the next Go minor releases or vendoring Go 1.18.10 encoding/xml (made the Excelize require minimum Go version to Go 1.18).

**Comment by fifsky** [comment]
Agreed to upgrade the minimum version to go 1.18

**Comment by tomtwinkle** [comment]
@xuri 
I'm using StreamWriter.
It seems to be working fine at the moment but does StreamWriter have the same problem?

**Comment by xuri** [comment]
> @xuri
> I'm using StreamWriter.
> It seems to be working fine at the moment but does StreamWriter have the same problem?

Yes, I suggest using the Go 1.20.7 and previous Go released version or waiting for the next Go minor releases.

**Comment by Abzaek** [comment]
I am encountering the same issue. SaveAs and SetCellValue don't return any error

**Comment by Abzaek** [comment]
Not working on "go version go1.22.6 linux/amd64" SaveAs and SetCellValue are not returning anything

**Comment by xuri** [comment]
Could you show us a complete, standalone example program or reproducible demo?
